File: CSCI-240/MultiUserApp/src/Address.java
import java.io.Serializable;

// Address.java
// contains information related to address
// Naeem Tai
// 4/20/2013

public class Address implements Serializable{

	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;
	
	// declare variables
	private String adrLineOne;
	private String adrLineTwo;
	private String city;
	private String state;
	private int zip;
	private boolean visible;
	
	// constructor with no parameters
	public Address(){
		this.adrLineOne = "";
		this.adrLineTwo = "";
		this.city = "";
		this.state = "";
		this.zip = 0;
		this.visible = false;
	}// end constructor
	
	// constructor with parameters
	public Address(String al1, String al2, String c, String s, int z, boolean v){
		this.adrLineOne = al1;
		this.adrLineTwo = al2;
		this.city = c;
		this.state = s;
		this.zip = z;
		this.visible = v;
	}// end constructor
	
	// return address to be printed
	public String printAddress(){
		String address = "";
		// check for empty spaces for display consistency
		if (this.visible){
			if (!this.adrLineOne.equals(""))
				address += "\n     " + this.adrLineOne + "\n";
			if (!this.adrLineTwo.equals(""))
				address += "     " + this.adrLineTwo + "\n";
			if (!this.city.equals(""))
				address += "     " + this.city + "\n";
			if (!this.state.equals(""))
				address += "     " + this.state + "\n";
			if (this.zip != 0)
				address += "     " + this.zip;
			return address;
		}
		else{
			return "Hidden";
		}
	}
	
	
	// getters
	
	// get line one of address
	public String getAdrLineOne(){
		return this.adrLineOne;
	}
	
	// get line two of address
	public String getAdrLineTwo(){
		return this.adrLineTwo;
	}
	
	// get city of address
	public String getCity(){
		return this.city;
	}
	
	// get state of address
	public String getState(){
		return this.state;
	}
	
	// get zip of address
	public int getZip(){
		return this.zip;
	}
	
	// get if address is visible or not
	public boolean isVisible(){
		return this.visible;
	}
	

	// setters
	
	// set line one of address
	public void setAdrLineOne(String al1){
		this.adrLineOne = al1;
	}
	
	// set line two of address
	public void setAdrLineTwo(String al2){
		this.adrLineTwo = al2;
	}
	
	// set city of address
	public void setCity(String c){
		this.city = c;
	}
	
	// set state of address
	public void setState(String s){
		this.state = s;
	}
	
	// set zip of address
	public void setZip(int z){
		this.zip = z;
	}
	
	// set if address is visible or not
	public void setVisible(boolean v){
		this.visible = v;
	}

}// end class

File: CSCI-240/Students/Address.cpp
// Address.cpp
// Naeem Tai
/* contains address data */

#include "Address.h"
#include <iostream>
#include <sstream>

using namespace std;

// empty constructor
// set adress data to unknown
Address::Address()
{
    adrLineOne = "# Anonymus Street";
    adrLineTwo = "Park Name";
    city = "Unknown City";
    state = "Unknown State";
    zip = 12345;
}

// copy constructor
Address::Address(const Address& original){
    adrLineOne = original.adrLineOne;
    adrLineTwo = original.adrLineTwo;
    city = original.city;
    state = original.state;
    zip = original.zip;
}

// custom constructor
// set address data as given
Address::Address(string al1, string al2, string c, string s, int z){
    adrLineOne = al1;
    adrLineTwo = al2;
    city = c;
    state = s;
    zip = z;
}


/* setters */

// set line one of address
void Address::setAdrLineOne(string al1){
    adrLineOne = al1;
}

// set line two of address
void Address::setAdrLineTwo(string al2){
    adrLineTwo = al2;
}

// set city of Address
void Address::setCity(string c){
    city = c;
}

// set state of Address
void Address::setState(string s){
    state = s;
}

// set zip code of address
void Address::setZip(int z){
    zip = z;
}


/* getters */

// get line one of address
string Address::getAdrLineOne(){
    return adrLineOne;
}

// get line two of address
string Address::getAdrLineTwo(){
    return adrLineTwo;
}

// get city of address
string Address::getCity(){
    return city;
}

// get state of address
string Address::getState(){
    return state;
}

// get zip code of address
int Address::getZip(){
    return zip;
}


/* additional methods */

// print address
void Address::printAddress(){
    cout << adrLineOne << ", " << adrLineTwo << ", " << city << ", " << state << ", " << zip << endl;
}

// return address data as a string
string Address::getAddress(){
    stringstream a;
    a << adrLineOne << " " << adrLineTwo << " " << city << " " << state << " " << zip;
    return a.str();
}
